### PR TITLE
Add LoadingBar component

### LINKED ISSUE #75: Progressive Loading Bar
- Visually shows the the merchant the time left to complete a task. (Either percentage or per task “1/12”)
- Used for uploading or downloading (processing of data)

### PR DESCRIPTION
**Pull request by connor-baer**
**Changes**

- Add progressive LoadingBar component
- Add util to generate unique ids

Closes #75.

### CODE REVIEW

**Comment by connor-baer** [comment]
Not sure why that one test is failing — all tests pass locally.

File: src/components/LoadingBar/LoadingBar.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+const baseStyles = ({ theme }) => css`
+  label: loading-bar;
+  display: flex;
+  align-items: center;
+  margin-bottom: ${theme.spacings.mega};
+`;
+
+const progressStyles = ({ theme, size, value, max }) => css`
```
Comment (felixjung) [inline src/components/LoadingBar/LoadingBar.js]: Let's store the border radius in a variable so you can dynamically calculate the inner radius.

File: src/components/LoadingBar/LoadingBar.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  }
+`;
+
+const labelStyles = ({ theme }) => css`
+  label: loading-bar__label;
+  ${textKilo({ theme })};
+  margin-left: ${theme.spacings.byte};
+`;
+
+const LoadingBarWrapper = styled('div')(baseStyles);
```
Comment (felixjung) [inline src/components/LoadingBar/LoadingBar.js]: I've started to prefix the base styles when there are multiple styled components that make up the main component. This way its clear which styled component the style function belongs to. Here I would probably call this `wrapperBaseStyles`. No need to add the remaining prefix (`LoadingBarWrapperStyles` would be kind of bananas). Maybe there is a better approach?

File: src/components/LoadingBar/LoadingBar.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * Loading bar component to indicate progress
+ */
+const LoadingBar = ({ children, max, value, ...props }) => {
+  const ariaId = id('loadingBar_');
+  return (
+    <LoadingBarWrapper>
+      <LoadingBarProgress
+        role="progressbar"
```
Comment (felixjung) [inline src/components/LoadingBar/LoadingBar.js]: Yay for all the a11y effort. 👏

File: src/util/unique-id.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// NOTE: Related issue https://github.com/facebook/react/issues/5867
```
Comment (felixjung) [inline src/util/unique-id.js]: 👍 up for those references.

**Pull request merged**